presto-tests/src/test/java/com/facebook/presto/tests/TestServerInfoResource.java

```diff
@@ -63,100 +69,90 @@ public void shutdown()
     public void testServerActive()
             throws Exception
     {
-        try (DistributedQueryRunner queryRunner = createQueryRunner(TINY_SESSION, ImmutableMap.of())) {
-            TestingPrestoServer coordinator = queryRunner.getServers()
-                    .stream()
-                    .filter(TestingPrestoServer::isCoordinator)
-                    .findFirst()
-                    .get();
-
-            ServerInfoResource serverInfoResource = coordinator.getServerInfoResource();
-            NodeState nodeState = serverInfoResource.getServerState();
-            assertTrue(nodeState == NodeState.ACTIVE);
-        }
+        TestingPrestoServer coordinator = queryRunner.getServers()
+                .stream()
+                .filter(TestingPrestoServer::isCoordinator)
+                .findFirst()
+                .get();
+
+        ServerInfoResource serverInfoResource = coordinator.getServerInfoResource();
+        NodeState nodeState = serverInfoResource.getServerState();
+        assertTrue(nodeState == NodeState.ACTIVE);
     }
 
     @Test(timeOut = SHUTDOWN_TIMEOUT_MILLIS)
     public void testServerInactiveThenActive()
             throws Exception
     {
-        try (DistributedQueryRunner queryRunner = createQueryRunner(TINY_SESSION, ImmutableMap.of())) {
-            TestingPrestoServer coordinator = queryRunner.getServers()
-                    .stream()
-                    .filter(TestingPrestoServer::isCoordinator)
-                    .findFirst()
-                    .get();
-
-            ServerInfoResource serverInfoResource = coordinator.getServerInfoResource();
-            Response response = serverInfoResource.updateState(NodeState.INACTIVE);
-            assertEquals(response.getStatus(), 200);
-            NodeState nodeState = serverInfoResource.getServerState();
-            assertTrue(nodeState == NodeState.INACTIVE);
-            response = serverInfoResource.updateState(NodeState.ACTIVE);
-            assertEquals(response.getStatus(), 200);
-            nodeState = serverInfoResource.getServerState();
-            assertTrue(nodeState == NodeState.ACTIVE);
-        }
+        TestingPrestoServer coordinator = queryRunner.getServers()
+                .stream()
+                .filter(TestingPrestoServer::isCoordinator)
+                .findFirst()
+                .get();
+
+        ServerInfoResource serverInfoResource = coordinator.getServerInfoResource();
+        Response response = serverInfoResource.updateState(NodeState.INACTIVE);
```

Comment:
The getServers copies of the server instances. I think it should be fine.

Reply:
Actually it is not a concern here since we are using `singleThreaded=true`.
https://github.com/prestodb/presto/pull/18027/files#diff-97784a051d25d4e1e14c144d22702a9a8a6ca0cf3e3972673dc909629dee1235R38